### PR TITLE
Display hint checks

### PR DESCRIPTION
**Pull request by srinijay98**
# COMPLETES #< INSERT LINK TO ISSUE >
https://jira-eng-gpk2.cisco.com/jira/browse/SPARK-354097
## This pull request addresses

< DESCRIBE THE CONTEXT OF THE ISSUE >
meeting display hints are set to true even when one has joined the meeting.
## by making the following changes
< DESCRIBE YOUR CHANGES >
Added a check to see if participant is in the meeting.

### Change Type

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Documentation update
- [ ] Tooling change
- [ ] Internal code refactor

## The following scenarios where tested

< ENUMERATE TESTS PERFORMED, WHETHER MANUAL OR AUTOMATED >

### I certified that

- [ ] I have read and followed [contributing guidelines](https://github.com/webex/webex-js-sdk/blob/master/CONTRIBUTING.md#submitting-a-pull-request)
- [ ] I discussed changes with code owners prior to submitting this pull request

- [ ] I have not skipped any automated checks
- [ ] All existing and new tests passed
- [ ] I have updated the documentation accordingly

---

Make sure to have followed the [contributing guidelines](https://github.com/webex/webex-js-sdk/blob/master/CONTRIBUTING.md#submitting-a-pull-request) before submitting.

### CODE REVIEW

File: packages/node_modules/@webex/plugin-meetings/src/locus-info/index.js [hunk 1]
```diff
@@ -779,14 +779,15 @@ export default class LocusInfo extends EventsScope {
...
    * @returns {undefined}
    * @memberof LocusInfo
    */
-  updateMeetingInfo(info, self) {
+  updateMeetingInfo(info, self, meetingState) {
     if (info && (!isEqual(this.info, info))) {
       const roles = self ? SelfUtils.getRoles(self) : this.parsedLocus.self?.roles || [];
 
-      const parsedInfo = InfoUtils.getInfos(this.parsedLocus.info, info, roles);
+      const parsedInfo = InfoUtils.getInfos(this.parsedLocus.info, info, roles, meetingState === MEETING_STATE.STATES.JOINED);
```
Comment (Coread) [inline packages/node_modules/@webex/plugin-meetings/src/locus-info/index.js:790]: question: Is this the actual constant you need here? There are two constants that both correspond to the string "JOINED". In the selfUtils the comparisons against state are made using the underscores _JOINED_

Comment (Coread) [inline packages/node_modules/@webex/plugin-meetings/src/locus-info/index.js:790]: suggestion: I think that this comparison is better living in the SelfUtils. Given that meetingState is from parsedLocus.self you can add the necessarily comparison function as a util in SelfUtils. I don't think it is necessary to pass the meetingState in separately as you already have self

File: packages/node_modules/@webex/plugin-meetings/src/locus-info/infoUtils.js [hunk 2]
```diff
@@ -3,9 +3,9 @@ import {SELF_ROLES} from '../constants';
 
 const InfoUtils = {};
 
-InfoUtils.parse = (info, roles) => {
+InfoUtils.parse = (info, roles, isJoined) => {
   const parsed = {
-    policy: InfoUtils.parsePolicy(info),
+    policy: InfoUtils.parsePolicy(info, isJoined),
```
Comment (Coread) [inline packages/node_modules/@webex/plugin-meetings/src/locus-info/infoUtils.js:8]: issue: this isn't quite right. We always want to parse the roles correctly. We should use isJoined when we define userDisplayHints. Currently it always includes parsed.policy, but it should only do that when the user isJoined.

File: packages/node_modules/@webex/plugin-meetings/src/locus-info/index.js [hunk 3]
```diff
@@ -795,14 +795,16 @@ export default class LocusInfo extends EventsScope {
   /**
    * @param {Object} info
    * @param {Object} self
+   * @param {string} meetingState
```
Comment (Coread) [inline packages/node_modules/@webex/plugin-meetings/src/locus-info/index.js:798]: you don't need this param anymore

Comment (srinijay98) [reply packages/node_modules/@webex/plugin-meetings/src/locus-info/index.js]: removed

File: packages/node_modules/@webex/plugin-meetings/src/locus-info/infoUtils.js (hunk 2 above)
Comment (srinijay98) [reply packages/node_modules/@webex/plugin-meetings/src/locus-info/infoUtils.js]: modified

File: packages/node_modules/@webex/plugin-meetings/src/locus-info/index.js (hunk 1 above)
Comment (srinijay98) [reply packages/node_modules/@webex/plugin-meetings/src/locus-info/index.js]: modified

**Review by Coread — APPROVED** [review]

**Review by arun3528 — CHANGES_REQUESTED** [review]
Can we check on this https://github.com/webex/webex-js-sdk/blob/master/packages/node_modules/%40webex/plugin-meetings/src/meeting/index.js#L1431

we can use this and dont need join code

**Comment by cscheffauer** [comment]
@arun3528 @srinijay98 can we continue on this and get this in? this is blocking our raise hand feature from getting completed. thank u!

**Review by arun3528 — APPROVED** [review]
Looks good to merge

**Comment by srinijay98** [comment]
Please check https://github.com/webex/webex-js-sdk/pull/2420/files

**Pull request closed without merging**